deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/FileUploadCompletionNotification.java

```
@@ -0,0 +1,195 @@
+     *
+     * @param correlationId the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     *                      for this upload from IoT Hub. This field is mandatory. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param isSuccess whether the file was uploaded successfully. This field is mandatory.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Create an instance of the FileUploadCompletionNotification for a single file upload operation using Azure Storage.
+     *
+     * @param correlationId the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     *                      for this upload from IoT Hub. This field is mandatory. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param isSuccess whether the file was uploaded successfully. This field is mandatory.
+     * @param statusCode is the status for the upload of the file to storage.
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess, Integer statusCode, String statusDescription)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Construct this notification with json
+     * @param json the json to parse.
+     */
+    public FileUploadCompletionNotification(String json)
+    {
```

do we need a `throws IllegalArgumentException` for thie constructor?

---

No, because IllegalArgumentException is a runtime exception, so java methods don't need to declare them. https://www.tutorialspoint.com/how-to-handle-the-runtime-exception-in-java#:~:text=The Runtime Exception is the,Runtime Exceptions are never checked.